### DISCUSSION TITLE
With the blog plugin, can I suppress the "continue reading" link?

### DISCUSSION THREAD

**Discussion by pitosalas** [squidfunk/mkdocs-material#6642] [Q&A]
In other words, I would like the whole post to appear on the index page?

**Comment by alexvoss** [comment]
That should be the behavior when you simply don't define an excerpt: https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#adding-an-excerpt

**Comment by pitosalas** [comment]
Not my experience. Here is an example:

---
title: "Top Python libraries of 2023"
author: Pito Salas
date: 2024-01-10
tags:
- ai
- data-science
- libraries
- package
- python-tag
---

[ Top Python libraries of 2023](< https://tryolabs.com/blog/top-python-
libraries-2023>) –Exploring the expansive universe of Python libraries in
2023, our curated selection goes beyond the usual suspects to uncover hidden
gems and innovative tools. Dive in!

Gives the following (note the “continue reading”.) Maybe I misunderstand,
maybe that’s not an excerpt but something else.


But basically I want not to have continue reading in any case. I want the
whole blog post plus meta data to be part of the index page. This is what
the singleton blog post looks like (what I see when I click continue
reading.) I would like that to be part of the blog index page.

**Comment by alexvoss** [comment]
I see, the "continue reading" link still exists on the index page. I guess it does make sense to have a link to the blog post itself, even if it is possible to read it in full on the index page. Perhaps a different link text might be better but I am not sure I can suggest something that would work in all cases?

In any case, you should be able to hide the link with some [extra css](https://squidfunk.github.io/mkdocs-material/customization/?h=extra+css#additional-css):

```css
.md-post__action {
  display: none;
}
```

**Comment by pitosalas** [comment]
Thanks… yeah, actually what I am trying is that all the text of the post
appear in the index page, no matter how long It is. Maybe someone else on
the list has an idea…?

**Reply by alexvoss** [reply]
I am not sure I understand what it is that you are trying to do that is not working. If I create a blog and do not define an excerpt, the whole post appears in the blog index:

<img width="1090" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/4134224/de13b1e3-9435-4bf6-8900-a4f5bc313afb">

I did check with the public version as well as Insiders. Can you please make a screenshot or provide a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) so we can see what is not working?

**Comment by pitosalas** [comment]
Hi Alexander


Absolutely!

Here’s my yaml:

site_name: Salas.com
site_description: Pito Salas Link Blog and More
site_author: Pito Salas
copyright: Copyright &copy 2000-2024 Pito Salas
docs_dir: docs
theme:
  name: material
  features:
    - navigation.footer
    - navigation.prefetch
    - navigation.tabs
    - search.suggest
  custom_dir: overrides

plugins:
  - search
  - blog:
      blog_dir: .
  - tags:
      enabled: true
      tags: true
      tags_file: tags.md
nav:
  - Blog:
    - index.md
  - Finding a Job: pages/finding-a-job.md
  - Independent Study: pages/independent-study.md
  - pages/keeping-up.md
  - pages/curacao.md
  - pages/recommendations.md
  - pages/about.md
  - tags.md
extra_css:
  - mystyle.css

Here is the screenshot, notice the “continue reading”.




Here’s what the linked page looks like. Notice it is identical except that
it shows the tags and the metadata.


On J

Here’s the markdown file:

---
title: "Top Python libraries of 2023"
author: Pito Salas
date: 2024-01-10
tags:
    - ai
    - data-science
    - libraries
    - package
    - python-tag
---

[ Top Python libraries of 2023](< https://tryolabs.com/blog/top-python-
libraries-2023>) –Exploring the expansive universe of Python libraries in
2023, our curated selection goes beyond the usual suspects to uncover hidden
gems and innovative tools. Dive in!

What I would like is that the index page (my first image with the continue
reading) instead showed a section like my second image, with the tags.

So in other words, don’t put a “continue reading” and don’t cut off the
post if it’s too long.

Thanks,

Pito

**Reply by squidfunk** [reply]
Please read our guide how to [create a minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/). Without it, we are not able to help you.

**Comment by pitosalas** [comment]
Here you go.

**Reply by alexvoss** [reply]
Hi, I am afraid I cannot see the zip file.

**Reply by ddorian** [reply]
I'm getting the same thing as him, but I will wait for him to upload his zip.

**Comment by pitosalas** [comment]
materia-repro.zip
<https://drive.google.com/file/d/1FsiAw8rClBtcHGPuYivRRqQ76I99B91h/view?usp=drive_web>

**Reply by alexvoss** [reply]
ok, next time please use the process for creating a minimal reproduction. We do not need the whole venv but we do need the information the info plugin spits out. That makes it small enough to just upload to Github Discussions. I removed the venv and compiled the result with the public version of Material. The whole post is visible. 

<img width="1082" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/4134224/843ea976-058f-4694-a5e1-cd2a365e971c">

@pitosalas and @ddorian, what is the behaviour you expected? I am still not clear about that. I can get that you may want to hide the "continue reading" link but we covered that.

**Reply by ddorian** [reply]
I think it was "To not print `Continue reading` link ?

**Comment by pitosalas** [comment]
Hi again

I think I sent you exactly that. The reproduction was done and a zip was
sent this morning.

But anyway, let’s forget it. I don’t think what I want is possible anyway.

Ok to close the ticket.

Thanks for your help,

Pito

I think it was "To not print Continue reading link ?

**Reply by alexvoss** [reply]
I am afraid I do not understand *what* it is that you need. Just concluding that it is not possible seems somewhat unsatisfying but if you want to please go ahead. 

The .zip file contained a whole venv, which is why it was 35MB, most of which I discard because I won't just run a venv from a zip file I downloaded on my machine (hope you understand). 

The [info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/) that is part of the instructions for creating a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) helps us to streamline the process of investigating issues. It is important for us that everyone use this in discussions and issues.

**Comment by pitosalas** [comment]
Thanks.
